src/edwards.rs

```diff
@@ -2,6 +2,70 @@
 //! Edwards Point operation implementations and definitions.
 //! Encoding/decoding processes implementations
 //! and support for all kind of interactions with them.
+//! 
+//! # Examples
+//! ```rust
+//! use zerocaf::edwards::*;
+//! use zerocaf::traits::ops::*;
+//! use zerocaf::traits::Identity;
+//! use zerocaf::field::FieldElement;
+//! use zerocaf::scalar::Scalar;
+//! 
+//! use subtle::Choice;
+//! use core::ops::{Add, Sub, Mul};
+//! 
+//! // You can work in different point coordinates like
+//! // Affine, Projective and Extended ones.
+//! // 
+//! // It's highly recommended to work over Extended ones
+//! // since are the fastest. And the ones you can compress
+//! // directly.
+//! 
+//! // In order to get a Point over the Doppio curve, 
+//! // you can do the following: 
+//! 
+//! // From a y-coordinate of a point: 
+//! let y = FieldElement([1799957170131195, 4493955741554471, 4409493758224495, 3389415867291423, 16342693473584]);
+//! // The `Choice` specifies the symbol that we want to get as a result
+//! // for the `x-coordinate`.
+//! let ex_point = EdwardsPoint::new_from_y_coord(&y, Choice::from(0u8)).unwrap();
+//! 
+//! // Create a random point. 
+//! let rngp = EdwardsPoint::new_random_point();
+//! 
+//! // Get it from an AffinePoint or a ProjectivePoint: 
+//! let exampl = EdwardsPoint::from(&ProjectivePoint::identity());
+//! 
+//! // The same examples work for the ProjectivePoint struct. 
+//! 
+//! // You can perform the following operations with an EdwardsPoint
+//! // or a ProjectivePoint: 
+//! 
+//! // Point addition:
+//! let p1 = &ex_point + &rngp;
+//! // Point subtraction (which is a point negation and a sum): 
+//! let p2 = &ex_point - &rngp;
+//! // Point doubling: 
+//! let p3 = &ex_point.double();
+//! // Scalar mul:
+//! let p4 = double_and_add(&ex_point, &Scalar::from(&8u8));
+//! 
+//! // You can also multiply by the co-factor directly: 
+//! assert!(p4 == mul_by_cofactor(&ex_point));
+//! 
+//! // In order to send points saving space, you can use
+//! // compressed points, repressented as: `CompressedEwdardsY`. 
```

Comment:
If you just push this section to master, I will re-write parts from master on my computer. Simply to touch up th explanations.

Reply:
Hmmm so merge this first? Okai I'll merge this first. Anyway, #71 should be also merged. You can rebase it.